File: data/EXP01/golang_no_system_var/2452/2452.output_2.go
func twoEditWords(queries []string, dictionary []string) []string {
    matches := []string{}
    
    for _, query := range queries {
        if checkMatch(query, dictionary) {
            matches = append(matches, query)
        }
    }
    
    return matches
}

func checkMatch(query string, dictionary []string) bool {
    for _, word := range dictionary {
        if query == word {
            return true
        }
        
        if isOneEditAway(query, word) {
            return true
        }
        
        if isTwoEditsAway(query, word) {
            return true
        }
    }
    
    return false
}

func isOneEditAway(word1 string, word2 string) bool {
    count := 0
    
    for i := 0; i < len(word1); i++ {
        if word1[i] != word2[i] {
            count++
        }
        
        if count > 1 {
            return false
        }
    }
    
    return count == 1
}

func isTwoEditsAway(word1 string, word2 string) bool {
    edits := 0
    i := 0
    j := 0
    
    for i < len(word1) && j < len(word2) {
        if word1[i] != word2[j] {
            edits++
            if edits > 2 {
                return false
            }
        } else {
            i++
        }
        
        j++
    }
    
    return edits == 2
}

File: data/EXP01/python/2452/2452.output_2.py
# @lc app=leetcode id=2452 lang=python3
class Solution:
    def twoEditWords(self, queries: List[str], dictionary: List[str]) -> List[str]:
        def is_edit_distance_one(s: str, t: str) -> bool:
            if len(s) != len(t):
                return False
            count_diff = 0
            for i in range(len(s)):
                if s[i] != t[i]:
                    count_diff += 1
                    if count_diff > 1:
                        return False
            return count_diff == 1
        
        def is_edit_distance_two(s: str, t: str) -> bool:
            if len(s) != len(t):
                return False
            count_diff = 0
            for i in range(len(s)):
                if s[i] != t[i]:
                    count_diff += 1
                    if count_diff > 2:
                        return False
            return count_diff == 2
        
        result = []
        for query in queries:
            for word in dictionary:
                if query == word or is_edit_distance_one(query, word) or is_edit_distance_two(query, word):
                    result.append(query)
                    break
        return result